DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,106 @@
+
+# Developers Guide
+
+ The page is designed to give new developers general guidelines for
+ implementing code consistent with the VOTCA and c++ style and standard.
+
+## Resources
+
+ A good starting point is to take a look at the c++ standard. Though the code
+ has not always consistently followed the c++ standard we now make an
+ effort to really enforce it and follow best practices.
+
+ - [Best Practices1](https://www.gitbook.com/book/lefticus/cpp-best-practices/details)
+ - [Best Practices2](https://google.github.io/styleguide/cppguide.html)
+
+## Testing
+
+ Each repository contains a src folder. Within the src folder exists a 
+ libraryfolder: libtools, libcsg etc... and a tools folder. A tests folder
+ should also exist in the src folder. If it does not you should create one.
+
+ For every new object and algorithm created there should exist a test. We
+ will be using the Boost libraries testing framework. Good documentation can
+ be found here:
+
+ - [Boost link](https://www.ibm.com/developerworks/aix/library/au-ctools1_boost/)
+ 
+ We will outline the general workflow here using the vec object in 
+ votca::tools. This object only has a header file it is in:
+ /tools/include/votca/tools/vec.h
+ 
+ Determine if a tests folder has already been created or not in /src if it
+ has not take a look at what was done in the votca-tools repo. 
+
+ 1. Create a test file in tools/src/tests/test_vec.cc must have the same name as
+    what appears in the foreach in the CMakeLists.txt file. And place the following 
+    contents
+
+```  
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE vec_test
+#include <boost/test/unit_test.hpp>
+#include <exception>
+
+#include <votca/tools/vec.h>
+
+using namespace std;
+using namespace votca::tools;
+
+BOOST_AUTO_TEST_SUITE(vec_test)
+
+
+BOOST_AUTO_TEST_CASE(test1){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+  :
+}
+BOOST_AUTO_TEST_CASE(test2){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+  :
+}
+:
+BOOST_AUTO_TEST_SUITE_END()
+```
+ 
+Replace the '...' and ':' with the appropriate syntax. 
+For more info on which boost test macros to use refer to the boost documentation
+
+ 2. To compile and test the code create a folder tools/build and run the following
+    commands: 
+
+```
+cmake -DENABLE_TESTING=ON ../
+make 
+make test
+```
+
+Ensure you have an up to date version of cmake or use cmake3 
+
+## CPP Codeing Style Guide
+
+ VOTCA now uses the clang formatter to automatically ensure consistent style.
+ The style is essentially the same, as what is used by google, with the 
+ exception that types and = are aligned to make the code more readable.
```

Comment:
This kind of alignment is quite annoying when you change the name of the variables. Of course you can rerun clang-format but during the edition, you already see that something with the formatting is wrong so you start to fix it manually... personally, I do not like it.

Reply:
I never considered that, but I agree with @MrTheodor.